server.js

```diff
@@ -2,34 +2,236 @@ import express from "express";
 import cors from "cors";
 import mongoose from "mongoose";
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
+import booksData from "./data/books.json";
+import { resolveShowConfigPath } from "@babel/core/lib/config/files";
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
 mongoose.Promise = Promise;
 
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
+const User = mongoose.model("User", {
+  name: String,
+  age: Number,
+  deceased: Boolean
+});
+
+
+const Book = mongoose.model("Book", {
+  bookID: Number,
+  title: String,
+  authors: String,
+  average_rating: Number,
+  isbn: Number,
+  isbn13: Number,
+  language_code: String,
+  num_pages: Number,
+  ratings_count: Number,
+  text_reviews_count: Number
+})
+
+if(process.env.RESET_DB) {
+  const resetDataBase = async () => {
+    await Book.deleteMany();
+    booksData.forEach(singleBook => {
+      const newBook = new Book(singleBook);
+      newBook.save();
+    })
+  }
+  resetDataBase();
+}
+
 const port = process.env.PORT || 8080;
 const app = express();
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send("This is a book API!");
 });
 
-// Start the server
+  app.get("/books/", async (req, res) => {
+    const Books = await Book.find({})
+    res.status(200).json({
+      success: true,
+      body: Books
+    })
+  })
+
+app.get("/books/id/:id", async (req, res) => {
+  try {
+    const singleBook = await Book.findById(req.params.id);
+    if (singleBook) {
+      res.status(200).json({
+        success: true,
+        body: singleBook 
+       });
+    } else {
+      res.status(404).json({
+        success: false,
+        body: {
+          message: "Could not find the book"
+        } 
+       });
+    }
+  } catch(error) {
+    res.status(400).json({
+      success: false,
+      body: {
+        message: "Invalid id"
+      } 
+     });
+  }
+  });
+
+  app.get("/books/title/:title", async (req, res) => {
```

Comment:
Title and Author search works, but they are case sensitive, and will not work if title has special characters in it such as é or #. [Week 13 common pitfalls](https://www.notion.so/technigo/Express-API-72574dc0940b41158ddb4c98e37b713a) have some tips that maybe can help if you want to look into it 🙂